modules/core/src/main/java/org/apache/ignite/configuration/IgniteConfiguration.java

```diff
@@ -3570,6 +3578,35 @@ public IgniteConfiguration setSqlSchemas(String... sqlSchemas) {
         return this;
     }
 
+    /**
+     * <b>This is an experimental feature. Envronment awareness approac may be changed.</b>
+     * <p>
+     *
+     * Configured environment type.
```

Comment:
I suggest to expand javadoc here and describe that the environment type provides a hint to Ignite whether the Ignite instance is launched on a standalone machine or inside a virtualized environment. Describe what differences are important for virtualized environment (currently, NAT).